src/graphic/minimap_renderer.cc

```diff
@@ -78,6 +78,13 @@ inline RGBColor calc_minimap_color(const Widelands::EditorGameBase& egbase,
 			}
 		}
 
+		if (layers & MiniMapLayer::Ship && f.field->get_first_bob()) {
+			const Widelands::MapObjectType type = f.field->get_first_bob()->descr().type();
+			if (type == Widelands::MapObjectType::SHIP) {
```

Comment:
```suggestion
		if ((layers & MiniMapLayer::Ship) && f.field->get_first_bob()) {
			if (f.field->get_first_bob()->descr().type() == Widelands::MapObjectType::SHIP) {
```
Two suggestions, one for readability, one no need for a local variable.

---

In the case that a ship crosses paths with a duck or ferry, checking only the first bob here might cause the ship to briefly vanish from the minimap. Better iterate all bobs and break when any ship is found.